src/libraries/System.Text.Json/src/System/Text/Json/Serialization/MemberAccessor.cs

```diff
@@ -41,7 +41,7 @@ protected MethodInfo ImmutableDictionaryCreateRangeMethod(Type constructingType,
             return createRangeMethod.MakeGenericMethod(typeof(string), elementType);
         }
 
-        private MethodInfo FindImmutableCreateRangeMethod(Type constructingType)
+        private MethodInfo? FindImmutableCreateRangeMethod(Type constructingType)
```

Comment:
This should be non-nullable. The null-checks above are unreachable (hence the Debug.Fail).

```suggestion
        private MethodInfo FindImmutableCreateRangeMethod(Type constructingType)
```